### PR TITLE
feat: add overview page

### PR DESCRIPTION
**Pull request by Teja2045**
#458

### CODE REVIEW

File: frontend/src/features/common/commonSlice.js [hunk 1]
```diff
@@ -109,6 +128,29 @@ export const commonSlice = createSlice({
...
         state.tokensInfoState.info = {};
       });
+
+    builder
+      .addCase(getAllTokensPrice.pending, (state) => {
+        state.allTokensInfoState.status = "pending";
+        state.allTokensInfoState.error = "";
+      })
+      .addCase(getAllTokensPrice.fulfilled, (state, action) => {
+        console.log("get", action.payload);
```
Comment (akhilkumarpilli) [inline frontend/src/features/common/commonSlice.js:138]:
```suggestion
```

File: frontend/src/features/common/commonSlice.js [hunk 2]
```diff
@@ -109,6 +128,29 @@ export const commonSlice = createSlice({
...
+        let data = action.payload.data || [];
+        const TokensPriceInfo = data.reduce((result, tokenInfo) => {
```
Comment (akhilkumarpilli) [inline frontend/src/features/common/commonSlice.js:140]:
```suggestion
        const tokensPriceInfo = data.reduce((result, tokenInfo) => {
```

File: frontend/src/pages/Home.js [hunk 3]
```diff
@@ -77,13 +89,18 @@ function getTabIndex(path) {
...
   const [network, setNetwork] = React.useState(selectedNetwork);
+  const networks = useSelector((state) => state.wallet.networks);
+  const wallet = useSelector((state) => state.wallet);
 
   const dispatch = useDispatch();
   const navigate = useNavigate();
   const location = useLocation();
-  const page = location.pathname.split('/')?.[location.pathname.split('/')?.length - 1]
+  const page =
+    location.pathname.split("/")?.[location.pathname.split("/")?.length - 1];
```
Comment (akhilkumarpilli) [inline frontend/src/pages/Home.js:103]: Write as separate utility function if using in different files, if not define `location.pathname.split("/")` in separate variable and use it.

File: frontend/src/pages/OverviewPage.jsx [hunk 4]
```diff
@@ -1,7 +1,41 @@
...
+    if (!distributionReset && Object.keys(distribution).length > 0) {
+      setDistributionReset(true);
+    }
+  }, [distribution]);
+
+  useEffect(() => {
+    if (!tokenInfoReset && Object.keys(tokenInfo).length > 0) {
+      setTokenInfoReset(true);
+    }
+  }, [tokenInfo]);
```
Comment (akhilkumarpilli) [inline frontend/src/pages/OverviewPage.jsx:32]: Avoid multiple useEffect and use a single state variable checking all three conditions at once.

File: frontend/src/pages/general-overview/ChainDetails.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  const rewards = useSelector(
+    (state) => state.distribution.chains[chainID].delegatorRewards.totalRewards
+  );
+  const staked = useSelector(
+    (state) => state.staking.chains[chainID].delegations.totalStaked
+  );
+  const wallet = useSelector((state) => state.wallet);
+  const denom = wallet.networks[chainID].network.config.currencies[0].coinDenom;
+  const decimals =
+    wallet.networks[chainID].network.config.currencies[0].coinDecimals;
```
Comment (akhilkumarpilli) [inline frontend/src/pages/general-overview/ChainDetails.js:20]: What if any of the inner variables is undefined? Use `?` operator

File: frontend/src/pages/general-overview/ChainDetails.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+  return (
+    <>
+      {balance.length > 0 ? (
+        <StyledTableRow>
+          <StyledTableCell size="small">
+            <Box sx={{ display: "flex", alignItems: "center" }}>
+              <Avatar sx={{ width: 24, height: 24 }} />
```
Comment (akhilkumarpilli) [inline frontend/src/pages/general-overview/ChainDetails.js:28]: Don't we pass any avatar url here?

**Review by akhilkumarpilli — CHANGES_REQUESTED** [review]

**Review by akhilkumarpilli — APPROVED** [review]

**Review by aleem1314 — APPROVED** [review]

**Pull request merged**